### PR TITLE
fix: move RestAPI functionality to submodule

### PR DESCRIPTION
**Pull request by MatthewLemmond**
### Description
we are moving the restapi calls for this module into a submodule so the root level module can be consumed without configuring/utilizing the restapi provider

### Release required?

- [ ] No release
- [x] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

Moved `var.user_list_visibility` and `var.fs_validated` from the root level module to a submodule, in order to consume this submodule you can use `source = "terraform-ibm-modules/iam-account-settings/ibm//modules/restapi-options` along with the latest version.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,27 +26,30 @@ The module handles the following account settings:
...
-The module currently supports setting the following FSCloud requirements using the RestAPI provider until the IBM provider supports these settings:
+The module currently does not support setting the following FSCloud requirements using the IBM provider:
 
 - Check whether user list visibility restrictions are configured in IAM settings for the account owner
 - Check whether the Financial Services Validated setting is enabled in account settings
 
-Tracking issue with IBM provider -> https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4204
+Tracking issue with IBM provider -> <https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4204>
+
+If you need to manage these FSCloud requirements via Terraform, please see [this submodule](./modules/restapi-options) which uses the RestAPI provider to manage these settings.
```
Comment (ocofaigh) [inline README.md:42]: we need to call out that its using experimental unsupported apis

**Review by ocofaigh — CHANGES_REQUESTED** [review]
As per scrum, lets just call the submodule in the custom example

File: README.md [hunk 2]
```diff
@@ -26,27 +26,30 @@ The module handles the following account settings:
...
 
 <!-- Below content is automatically populated via pre-commit hook -->
 <!-- BEGIN OVERVIEW HOOK -->
 ## Overview
 * [terraform-ibm-iam-account-settings](#terraform-ibm-iam-account-settings)
+* [Submodules](./modules)
+    * [restapi-options](./modules/restapi-options)
```
Comment (ocofaigh) [inline README.md:49]: perhaps ask in the team channel for suggestions on the naming here. I was thinking something with the word "experimental" in it, but lets see what other think?

File: README.md [hunk 3]
```diff
@@ -26,25 +26,27 @@ The module handles the following account settings:
...
-The module currently supports setting the following FSCloud requirements using the RestAPI provider until the IBM provider supports these settings:
+The module currently does not support setting the following FSCloud requirements using the IBM provider:
 
 - Check whether user list visibility restrictions are configured in IAM settings for the account owner
 - Check whether the Financial Services Validated setting is enabled in account settings
 
-Tracking issue with IBM provider -> https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4204
+Tracking issue with IBM provider -> <https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4204>
+
+If you need to manage these FSCloud requirements via Terraform, please see [this submodule](./modules/restapi-options) which uses the RestAPI provider to manage these settings.
```
Comment (ocofaigh) [inline README.md:42]: hyperlink needs to be updated

File: README.md [hunk 4]
```diff
@@ -26,25 +26,27 @@ The module handles the following account settings:
...
 
 <!-- Below content is automatically populated via pre-commit hook -->
 <!-- BEGIN OVERVIEW HOOK -->
 ## Overview
 * [terraform-ibm-iam-account-settings](#terraform-ibm-iam-account-settings)
+* [Submodules](./modules)
+    * [experimental-extended-account-settings](./modules/experimental-extended-account-settings)
```
Comment (ocofaigh) [inline README.md:49]: `experimental-extended-account-settings` seems overkill. Its a submodule, so a person consuming is going to be consuming like this anyway: `"terraform-ibm-modules/iam-account-settings/ibm//modules/<submodule>"`

How about just calling it `experimental` ?

File: examples/custom/README.md [hunk 5]
```diff
@@ -1,3 +1,3 @@
 # Account Custom Settings example
 
-An example of custom settings being applied to the account. This use case may be helpful in development environments.
+An example of custom settings being applied to the account. This use case may be helpful in development environments. This example consumes the experimental options submodule that uses the RestAPI provider to call an experimental API that is currently unsupported.
```
Comment (ocofaigh) [inline examples/custom/README.md:3]: Can we be more specific as to the options the experimental submodule is going to set in the example?

File: modules/experimental-extended-account-settings/README.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
+# IAM Account Settings with RestAPI provider module
```
Comment (ocofaigh) [inline modules/experimental-extended-account-settings/README.md:1]: `# Experimental submodule`

File: modules/experimental-extended-account-settings/README.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+The module currently supports setting the following FSCloud requirements using the RestAPI provider until the IBM provider supports these settings:
+
+- Check whether user list visibility restrictions are configured in IAM settings for the account owner
+- Check whether the Financial Services Validated setting is enabled in account settings
+
+Tracking issue with IBM provider -> <https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4204>
+
+For further usage information on the module, please see the [base module readme](../../README.md#usage).
+
+Note: the API used for these settings is experimental and currently unsupported.
```
Comment (ocofaigh) [inline modules/experimental-extended-account-settings/README.md:40]: This needs to be cleaned up - it should only mention what the submodule does. Its also missing a usage block

**Review by ocofaigh — CHANGES_REQUESTED** [review]
Ohh wait a second - the submodule should ONLY support setting the experimental apis, not everything else the main module does. See it as an extension you can opt in ti

**Comment by MatthewLemmond** [comment]
@ocofaigh one note, I've added the restapi submodule to the fscloud profile and set the endpoint as private, but I wasn't sure if this submodule should be called there so please let me know if it should be removed or if anything within the submodule call should be changed.

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**